spdx/parsers/tagvaluebuilders.py

```diff
@@ -24,25 +23,22 @@
 from spdx import snippet
 from spdx import utils
 from spdx import version
+from spdx.checksum import Checksum
 
-from spdx.document import ExternalDocumentRef
+from spdx.document import ExternalDocumentRef, Document
 from spdx.package import PackagePurpose
 from spdx.parsers.builderexceptions import CardinalityError
 from spdx.parsers.builderexceptions import OrderError
 from spdx.parsers.builderexceptions import SPDXValueError
 from spdx.parsers import validations
 
 
-def checksum_from_sha1(value):
-    """
-    Return an spdx.checksum.Algorithm instance representing the SHA1
-    checksum or None if does not match CHECKSUM_RE.
-    """
-    # More constrained regex at lexer level
-    CHECKSUM_RE = re.compile("SHA1:\\s*([\\S]+)", re.UNICODE)
+def checksum_algorithm_from_string(value):
+    CHECKSUM_RE = re.compile("(ADLER32|BLAKE2b-256|BLAKE2b-384|BLAKE2b-512|BLAKE3|MD2|MD4|MD5|MD6|" \
+        "SHA1|SHA224|SHA256|SHA384|SHA512|SHA3-256|SHA3-384|SHA3-512):\\s*([a-fA-F0-9]*)")
```

Comment:
This method would make sense as part of the `ChecksumAlgorithm` class. But I'll leave it up to you whether you want to refactor this now

Reply:
moved to `ChecksumAlgorithm`